src/runtime_src/core/include/experimental/xrt_xclbin.h

```diff
@@ -33,13 +36,270 @@
 typedef void* xrtXclbinHandle;
 
 #ifdef __cplusplus
+
+// Utility to replicate pimpl
+template<typename IMPL>
+class Pimpl
+{
+public:
+  Pimpl() = default;
+
+  virtual ~Pimpl() = default;
+
+  explicit Pimpl(std::shared_ptr<IMPL> handle) : handle(std::move(handle)) {}
+
+  std::shared_ptr<IMPL>
```

Comment:
Let's change this to return
`const std::shared_ptr<ImplType>&`
to avoid the copying